docs/data-sources/azapi_operation.md

```diff
@@ -0,0 +1,110 @@
+---
+subcategory: ""
+layout: "azapi"
+page_title: "Generic Azure Resource Operation Data Source: azapi_operation"
+description: |-
+  Perform resource function to get information from an existing azure resource
+---
+
+# azapi_operation
+
+This resource can perform resource function to get information form any existing Azure resource manager resource.
+
+## Example Usage
+
+```hcl
+terraform {
+  required_providers {
+    azapi = {
+      source = "Azure/azapi"
+    }
+  }
+}
+
+provider "azapi" {
+}
+
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_resource_group" "example" {
+  name     = "example-rg"
+  location = "west europe"
+}
+
+resource "azurerm_automation_account" "example" {
+  name                = "example-account"
+  resource_group_name = azurerm_resource_group.example.name
+  location            = azurerm_resource_group.example.location
+  sku_name            = "Basic"
+}
+
+data "azapi_operation" "example" {
+  type                   = "Microsoft.Automation/automationAccounts@2021-06-22"
+  resource_id            = azurerm_automation_account.example.id
+  operation              = "listKeys"
+  response_export_values = ["*"]
+}
+```
+
+## Arguments Reference
+
+The following arguments are supported:
+
+* `type` - (Required) It is in a format like `<resource-type>@<api-version>`. `<resource-type>` is the Azure resource type, for example, `Microsoft.Storage/storageAccounts`.
+  `<api-version>` is version of the API used to manage this azure resource.
+
+* `resource_id` - (Required) The ID of an existing azure source.
+
+* `operation` - (Optional) The name of the resource operation. It's also possible to make Http requests towards the resource ID if leave this field empty.
+
+---
+* `body` - (Optional) A JSON object that contains the request body.
+
+* `method` - (Optional) Specifies the Http method of the azure resource operation. Allowed values are `POST`, `PATCH`, `GET`, `PUT`, `DELETE`, `CONNECT`, `HEAD`, `OPTIONS` and `TRACE`. Defaults to `POST`.
```

Comment:
Should we narrow down the allowed enums for this `method`, as it is only used for query?